api/api-secret-configs-v1/src/main/java/com/thoughtworks/go/apiv1/secretconfigs/representers/SecretConfigRepresenter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv1.secretconfigs.representers;
+
+import com.thoughtworks.go.api.base.OutputWriter;
+import com.thoughtworks.go.api.representers.ConfigurationPropertyRepresenter;
+import com.thoughtworks.go.api.representers.ErrorGetter;
+import com.thoughtworks.go.api.representers.JsonReader;
+import com.thoughtworks.go.config.SecretConfig;
+import com.thoughtworks.go.spark.Routes;
+import org.apache.commons.collections4.CollectionUtils;
+
+import java.util.Collections;
+import java.util.Map;
+
+public class SecretConfigRepresenter {
+    public static void toJSON(OutputWriter jsonWriter, SecretConfig secretConfig) {
+        if (secretConfig == null)
+            return;
+        jsonWriter.addLinks(linksWriter -> linksWriter
+                .addLink("self", Routes.SecretConfigsAPI.id(secretConfig.getId()))
+                .addAbsoluteLink("doc", Routes.SecretConfigsAPI.DOC)
+                .addLink("find", Routes.SecretConfigsAPI.find()))
+                .add("id", secretConfig.getId())
+                .add("plugin_id", secretConfig.getPluginId())
+                .addIfNotNull("description", secretConfig.getDescription());
+
+        if (secretConfig.hasErrors()) {
+            Map<String, String> fieldMapping = Collections.singletonMap("pluginId", "plugin_id");
+            jsonWriter.addChild("errors", errorWriter -> new ErrorGetter(fieldMapping).toJSON(errorWriter, secretConfig));
+        }
+
+        jsonWriter.addChildList("properties", listWriter -> {
```

Comment:
Not sure if `properties` is the right word for the configuration. I know that we are using it in other APIs. @maheshp  - WDYT?

Reply:
@bdpiparva for the sake of consistency I would keep it as `properties`.

Reply:
Keeping it as `properties`.